src/bitcoin/poller/looper.rs

```diff
@@ -46,44 +46,53 @@ fn update_coins(
             ..
         } = utxo;
         // We can only really treat them if we know the derivation index that was used.
-        let address = match address.require_network(network) {
-            Ok(addr) => addr,
-            Err(e) => {
-                log::error!("Invalid network for address: {}", e);
-                continue;
+        let (derivation_index, is_change) = match address {
+            UTxOAddress::Address(address) => {
+                let address = match address.require_network(network) {
+                    Ok(addr) => addr,
+                    Err(e) => {
+                        log::error!("Invalid network for address: {}", e);
+                        continue;
+                    }
+                };
+                if let Some((derivation_index, is_change)) =
+                    db_conn.derivation_index_by_address(&address)
+                {
+                    (derivation_index, is_change)
+                } else {
+                    // TODO: maybe we could try out something here? Like bruteforcing the next 200 indexes?
+                    log::error!(
+                        "Could not get derivation index for coin '{}' (address: '{}')",
+                        &utxo.outpoint,
+                        &address
+                    );
+                    continue;
+                }
             }
+            UTxOAddress::DerivIndex(index, is_change) => (index, is_change),
         };
-        if let Some((derivation_index, is_change)) = db_conn.derivation_index_by_address(&address) {
-            // First of if we are receiving coins that are beyond our next derivation index,
-            // adjust it.
-            if derivation_index > db_conn.receive_index() {
-                db_conn.set_receive_index(derivation_index, secp);
-            }
-            if derivation_index > db_conn.change_index() {
-                db_conn.set_change_index(derivation_index, secp);
-            }
+        // First of if we are receiving coins that are beyond our next derivation index,
+        // adjust it.
+        if derivation_index > db_conn.receive_index() {
+            db_conn.set_receive_index(derivation_index, secp);
+        }
+        if derivation_index > db_conn.change_index() {
+            db_conn.set_change_index(derivation_index, secp);
+        }
```

Comment:
shouldn.t we take in account `is_change`?
```suggestion
        // First of if we are receiving coins that are beyond our next derivation index,
        // adjust it.
        if derivation_index > db_conn.receive_index() && !is_change{
            db_conn.set_receive_index(derivation_index, secp);
        }
        if derivation_index > db_conn.change_index() && is_change {
            db_conn.set_change_index(derivation_index, secp);
        }
```

Reply:
I think you're right. I'll create an issue for this as this code has only been moved as part of this PR.

Reply:
Please see https://github.com/wizardsardine/liana/issues/1333.